cmd/main.go

```diff
@@ -395,3 +413,32 @@ func refreshCertPeriodically() {
 		}
 	}
 }
+
+// updatePodLabels updates the role label of pods based on the master election.
+func updatePodLabels(mgr manager.Manager) error {
+	c := mgr.GetClient()
+	// Fetch all pods in the given namespace
+	podList := &corev1.PodList{}
+	if err := c.List(context.TODO(), podList, &client.ListOptions{Namespace: nsxOperatorNamespace}); err != nil {
+		return fmt.Errorf("failed to list Pods in Namespace %s: %w", nsxOperatorNamespace, err)
+	}
+
+	// Iterate over the pods and update the role labels
+	for _, po := range podList.Items {
+		targetRole := roleStandby
+		if po.Name == nsxOperatorPodName {
+			targetRole = roleMaster
+		}
+
+		if po.Labels[roleKey] != targetRole {
```

Comment:
I'm not sure but the pod labels may be a nil map. Had better add a check to initialize pod.Labels if it is nil, ensuring no panic occurs when updating labels.

```
		if pod.Labels == nil {
			pod.Labels = make(map[string]string)
		}
...
```